### DISCUSSION TITLE
I can't get my saved card in user profile

### DISCUSSION THREAD

**Discussion by sentance** [swellstores/community#514] [Help] [answered]
Hello,

We are looking to enhance our user experience by implementing a new feature for managing payment methods within our shop. As part of this initiative, we've integrated a GraphQL query into our codebase to retrieve credit cards data
https://developers.swell.is/frontend-api/credit-card-management

However, despite implementing all the necessary functions to send requests, we are encountering an issue where the server consistently returns null

We would greatly appreciate your assistance in resolving this matter. Any insights, guidance, or support you can provide to help us debug and rectify this issue would be invaluable.

Thank you for your attention to this matter.

![Screenshot_32](https://github.com/swellstores/community/assets/10882664/49a903f3-275c-436a-b58b-87ea212dabde)
![Screenshot_31](https://github.com/swellstores/community/assets/10882664/6e0ed2c4-8aff-409e-be6c-a2049c77a33b)

**Comment by sentance** [comment]
and the credit card is in my account
![Screenshot_34](https://github.com/swellstores/community/assets/10882664/aacca96e-d622-43b0-a920-1bb6af4fc41a)

**Reply by logeshswell** [reply, marked as answer]
It's a gap in our graphQL implementation and we already received a request to support this feature.
You will receive an update if it becomes available.

**Reply by flonUK** [reply]
@logeshswell @ericingram could you please explain if this is inability to add/edit addresses and payments is specific to our store? 

You have a list of ecommerce stores, I have registered with some of these and yes you can add and edit payments and add addresses so why is this different for us? This is extremely weird to me, especially as it is not that urgent to deal with, so please tell me if I am missing something. Is this supposed to be a normal thing to fix whenever? 

As it stands, we will have to delete a customer's account so they can register and pay, obviously people simply drop out for good reason, no one has time for that in 2024. Do we just have the wrong set of customers that want to edit their address or payment? What am I missing?  What are others doing?

Address issue: https://github.com/orgs/swellstores/discussions/504

Thanks

**Reply by ericingram** [reply]
Swell.js has full support for this, I would recommend using that besides GraphQL. In general our GQL implementation has gaps currently and we are looking forward to revamping it entirely in the future, but it's not planned yet.

**Comment by sentance** [comment]
Do you have any instructions for this? How can user manage his payments methods?

**Reply by ericingram** [reply]
Here are some docs on managing credit card payment methods using swell.js https://developers.swell.is/frontend-api/credit-card-management

**Comment by sentance** [comment]
these methods returning null
That's why I created this topic

**Reply by ericingram** [reply]
Is the account logged in? They will return null if not

**Comment by sentance** [comment]
What wrong with this request?
![Screenshot_36](https://github.com/swellstores/community/assets/10882664/623aaec3-4098-46be-acc4-2b8830dc4cdc)
![Screenshot_37](https://github.com/swellstores/community/assets/10882664/297970b1-ed61-4205-b5e5-968c0888f5fe)
and the same answer (null) when I'm trying to get addresses or cards. 
But for example for subscriptions it works. 
When I try to invoke any methods without authorization it return me "error - unhandledRejection: Error: You must be logged in to perform this action" except these two - await swell.account.listCards() and await swell.account.listAddresses()
they returns null

**Reply by ericingram** [reply]
Sounds like there could be an issue with the two methods you mentioned, in that they should throw an error if not logged in.

The account.get method is supposed to return null when not logged in though.

Have you tried performing login before making those calls otherwise?

cc @logeshswell

**Comment by sentance** [comment]
Sure. Other methods works

**Comment by sentance** [comment]
![Screenshot_38](https://github.com/swellstores/community/assets/10882664/4b9bf856-44cf-48ba-a228-2b6183079960)
It returns me my orders from login method
![Screenshot_39](https://github.com/swellstores/community/assets/10882664/7215413d-aec3-44f7-8fde-2e5318f08cc5)
So there is a mess

**Reply by ericingram** [reply]
What’s the mess exactly? I can’t tell from your screenshot.

**Comment by logeshswell** [comment]
This issue is resolved. Please check and confirm if it's working fine now.
